main.cpp

```diff
@@ -121,196 +121,226 @@ struct CarWash
-    1) wash dishes
-    2) dry dishes
-    3) start at timer
- */
+
+std::string SodaMaker::getManufacturerName() {
+    return "SodaStream";
+}
+
+
+//--------------------------------------------------------
+// DishWasher
+//--------------------------------------------------------
 
  struct DishWasher
  {
-    // volume of washing room
     float washingRoomVolume = 5.2f;
-    // number of programs
     int numOfPrograms = 12;
-    // duration of duration of the washing process
-    int washingDuration = 120;
-    // amount of detergen tank
-    float amountDetergenTank = 0.200f;
-    // duration of timer
+    int maxWashingDuration = 120;
+    float maxWashTemperature = 90.0f;
     int timerDuration = 360;
+    int waitMilliSec = 0;
 
-    // wash dishes (returns true after the sucessful process)
     bool washDishes(int programNumber);
-    // dry dishes (returns true after the sucessful process)
-    bool dryDishes(int duration, int temperature);
-    // start at timer
+    bool dryDishes(int duration, float temperature);
     void startAtTimer(int startMinutesLater);   
  };
 
-/*
-Thing 4) Gas Grill
-5 properties:
-    1) size of grill plate
-    2) volume of coal
-    3) number level positions
-    4) heat temperature
-    5) port size of gas tank 
-3 things it can do:
-    1) grill sausage
-    2) light the flame
-    3) turn of the flame
- */
+
+bool DishWasher::washDishes(int programNumber)
+{
+    if(programNumber > 12) return false;
+    // Wash dishes!
+    return true;
+}
+
+
+bool DishWasher::dryDishes(int duration, float temperature)
+{
+    if(duration <= maxWashingDuration || temperature <= maxWashTemperature) 
+    {
+        // Drying!
+        return true;
+    }
+    return false;
+}
+
+
+void DishWasher::startAtTimer(int startMinutesLater)
+{
+    waitMilliSec = startMinutesLater * 1000;
+    // sleep(startMinutesLater * startMinutesLater);
+    DishWasher::washDishes(1);    
+} 
+
+//--------------------------------------------------------
+// GasGrill
+//--------------------------------------------------------
 
 struct GasGrill
 {
-    // size of grill plate
     double grillPlateSize = 50;
-    // volume of coal
     int volumeOfCoal = 5;
-    // number level positions
     int numLevelPositions = 3;
-    // heat temperature
     int heatTemperature = 1000;
-    // port size of gas tank 
     int gasTankPortSize = 5;
+    int distanceToCoal = 5;
+    float currentTemperature = 0.0f;
 
-    // grill sausage
     void grillSausage(int levelPosition);
-    // light the flame (returns true if the process was sucessful)
     bool ligthFlame();
-    // turn off the flame
     void turnOffFlame();
-
 };
 
 
-/*
-Thing 5) Display
-5 properties:
-    1) Color Depth
-    2) Pixel width
-    3) Pixel height
-    4) Power consumption
-    5) Brightness
-3 things it can do:
-    1) Display text
-    2) Display image
-    3) Turn backlight off 
- */
+void GasGrill::grillSausage(int levelPosition)
+{
+    GasGrill::ligthFlame();
```

Comment:
spell `light` correctly
remove the fully-qualified function name.  `lightFlame` is not a static member function
- [ ] revise